spec/models/subscription_spec.rb

```diff
@@ -149,4 +149,23 @@
       it { is_expected.to be_empty }
     end
   end
+
+  describe ".with_paid_transactions" do
+    let(:subscription_with_paid_transactions) { create(:subscription, :canceled) }
+    let(:subscription_without_paid_transactions) { create(:subscription, :pending_payment) }
+
+    subject { described_class.with_paid_transactions }
+
+    before do
+      [:refunded, :pending_payment, :refused, :processing].each do |status|
+        create(:transaction, status: status, subscription: subscription_without_paid_transactions)
```

Comment:
Why don't you use traits to set the status? You could just do `create(:transaction, status, subscription: subscription_without_paid_transactions)`